v2/internal/pkg/batch/concurrent_worker.go

```diff
@@ -0,0 +1,239 @@
+package batch
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/openshift/oc-mirror/v2/internal/pkg/api/v2alpha1"
+	clog "github.com/openshift/oc-mirror/v2/internal/pkg/log"
+	"github.com/openshift/oc-mirror/v2/internal/pkg/manifest"
+	"github.com/openshift/oc-mirror/v2/internal/pkg/mirror"
+	"github.com/vbauerster/mpb/v8"
+	"github.com/vbauerster/mpb/v8/decor"
+)
+
+func NewConcurrentBatch(log clog.PluggableLoggerInterface,
+	logsDir string,
+	mirror mirror.MirrorInterface,
+) BatchInterface {
+	copiedImages := v2alpha1.CollectorSchema{
+		AllImages: []v2alpha1.CopyImageSchema{},
+	}
+	return &ConcurrentBatch{Log: log, LogsDir: logsDir, Mirror: mirror, CopiedImages: copiedImages}
+}
+
+type ConcurrentBatch struct {
+	Log          clog.PluggableLoggerInterface
+	LogsDir      string
+	Mirror       mirror.MirrorInterface
+	Manifest     manifest.ManifestInterface
+	CopiedImages v2alpha1.CollectorSchema
+	Progress     *ProgressStruct
+}
+
+type BatchSchema struct {
+	BatchName string
+	Size      int
+	Images    v2alpha1.CollectorSchema
+}
+
+// Worker - the main batch processor
+func (o *ConcurrentBatch) Worker(ctx context.Context, collectorSchema v2alpha1.CollectorSchema, opts mirror.CopyOptions) (v2alpha1.CollectorSchema, error) {
+
+	var errArray []mirrorErrorSchema
+	var mu sync.Mutex
+	var wg errgroup.Group
+
+	var mirrorMsg string
+	if opts.Function == string(mirror.CopyMode) {
+		mirrorMsg = "copying"
+	} else if opts.Function == string(mirror.DeleteMode) {
+		mirrorMsg = "deleting"
+	}
+
+	startTime := time.Now()
+
+	batches := splitImagesToBatches(collectorSchema)
+
+	o.Log.Info("🚀 Start " + mirrorMsg + " the images...")
+
+	o.Log.Info("images to %s %d ", opts.Function, len(collectorSchema.AllImages))
+	o.Log.Debug("batch count %d ", len(batches))
+
+	for _, batch := range batches {
+
+		p := mpb.New()
+		total := len(batch.Images.AllImages)
+		for index, img := range batch.Images.AllImages {
+			spinnerText := fmt.Sprintf("%d/%d: %s ", index+1, total, img.Origin)
```

Comment:
Could we add the currentImage number against the total - ie. total release images 209 then 
` 10/209 ` 
`11/209` 
etc etc

also for operators and additional images 
I think the batch number and size are irrelevant

Reply:
Yes, good catch!
Done